src/hooks/accounting.ts

```diff
@@ -93,12 +93,13 @@ export const useAccounting = (): UseAccountingHook => {
     setIsloadingUncashed(true)
     const promises = settlements.settlements.settlements
       .filter(({ received }) => received.toBigNumber.gt('0'))
-      .map(({ peer }) => beeDebugApi.chequebook.getPeerLastCashout(peer))
+      .map(({ peer }) => makeRetriablePromise(() => beeDebugApi.chequebook.getPeerLastCashout(peer)))
 
-    Promise.all(promises)
-      .then(setUncashedAmounts)
-      .catch(setErr)
-      .finally(() => setIsloadingUncashed(false))
+    Promise.allSettled(promises).then(settlements => {
```

Comment:
So with the retries, this means that if there is repeatedly error, we have no results until all the retries are exhausted, right?

I am fine with it, just wanted to give you a food for thought.

Reply:
Yes, that is correct.

I am going to be spend at most an hour on making this a bit smoother, but if it is not doable in an easy way, I'd say it is not a big loss. 🙂

Reply:
I agree, thanks!